File: rx-codelab/src/main/java/com/jraska/rx/codelab/SchedulerProvider.java
package com.jraska.rx.codelab;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

public class SchedulerProvider {
  private final Scheduler main;
  private final Scheduler io;
  private final Scheduler computation;

  SchedulerProvider(Scheduler main, Scheduler io, Scheduler computation) {
    this.main = main;
    this.io = io;
    this.computation = computation;
  }

  public Scheduler getMain() {
    return main;
  }

  public Scheduler getIo() {
    return io;
  }

  public Scheduler getComputation() {
    return computation;
  }

  public static SchedulerProvider testSchedulers() {
    return new SchedulerProvider(Schedulers.trampoline(), Schedulers.trampoline(), Schedulers.trampoline());
  }
}

File: rx-codelab/src/main/java/com/jraska/rx/codelab/SchedulerProvider.kt
package com.jraska.rx.codelab

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers

class SchedulerProvider internal constructor(
  val main: Scheduler,
  val io: Scheduler,
  val computation: Scheduler
) {
  companion object {
    fun testSchedulers(): SchedulerProvider {
      return SchedulerProvider(Schedulers.trampoline(), Schedulers.trampoline(), Schedulers.trampoline())
    }

    fun realSchedulers(): SchedulerProvider {
      return SchedulerProvider(Schedulers.trampoline(), Schedulers.io(), Schedulers.computation())
    }
  }
}
